src/NuGet.Clients/NuGet.CommandLine/Commands/ProjectFactory.cs

```diff
@@ -290,6 +300,18 @@ public Packaging.PackageBuilder CreateBuilder(string basePath, NuGetVersion vers
                         string.Format(NuGetResources.ProjectJsonPack_Deprecated, builder.Id),
                         NuGetLogCode.NU5126));
                 _usingJsonFile = true;
+
+                _ = bool.TryParse(_environmentVariableReader.GetEnvironmentVariable(NUGET_ENABLE_LEGACY_PROJECT_JSON_PACK), out bool enableLegacyProjectJsonPack);
+
+                if (!enableLegacyProjectJsonPack)
+                {
+                    Logger.Log(
+                        PackagingLogMessage.CreateError(
+                            string.Format(NuGetResources.Error_ProjectJson_Deprecated_And_Removed, builder.Id, NUGET_ENABLE_LEGACY_PROJECT_JSON_PACK),
+                            NuGetLogCode.NU5041));
```

Comment:
I believe you wanted to use newly created NU5042 or NU5043 here.

Reply:
NU5042. Fixed.